test_runner/src/main/kotlin/ftl/args/AndroidTestShard.kt

```diff
@@ -10,21 +10,38 @@ import ftl.util.FlankTestMethod
 import java.io.File
 
 object AndroidTestShard {
+    private fun getTestApks(args: AndroidArgs, origTestApks: List<String>): List<String> {
+        val testApks = if (origTestApks.isNotEmpty()) {
+            origTestApks
+        } else {
+            val allTestApks = mutableListOf<String>()
+            val mainTestApk = args.testApk
+            if (mainTestApk != null) allTestApks.add(mainTestApk)
+            allTestApks.addAll(args.additionalAppTestApks.map { it.test })
+            allTestApks
+        }
 
-    // computed properties not specified in yaml
-    fun getTestShardChunks(args: AndroidArgs, testApk: String): ShardChunks {
-        // Download test APK if necessary so it can be used to validate test methods
-        val testLocalApk = if (testApk.startsWith(FtlConstants.GCS_PREFIX))
-            GcStorage.download(testApk) else
-            testApk
+        return testApks.map { testApk ->
+            if (testApk.startsWith(FtlConstants.GCS_PREFIX)) GcStorage.download(testApk) else
+                testApk
+        }
+    }
 
-        val filteredTests = getTestMethods(args, testLocalApk)
+    fun getTestShardChunks(args: AndroidArgs, testApks: List<String> = listOf()): ShardChunks {
+        val resolvedApks = getTestApks(args, testApks)
+        val filteredTests = resolvedApks.map { getTestMethods(args, it) }.flatten()
```

Comment:
I think this is producing wrong results. Here we have list of all tests from all test pairs. With normal flow we have new matrix for each test pair and inside that matrix flank divides test into shards. With this solution we have all tests in one list (from all test pairs) and then flank make shards, so we basically neglect how tests are divided among matrices. I may be wrong, but this is not result we would like to have.

Reply:
Thanks. I think you're right.

I think there are two different use cases:
- Serialize the shards to `android_shards.json` for debugging. 
- Given a single test apk, generate shards.

Maybe these should be different methods after all.